packages/react/src/components/Table/Table.test.jsx

```diff
@@ -590,27 +592,40 @@ describe('Table', () => {
         view={{
           toolbar: {
             search: {
-              defaultValue: '',
+              defaultValue: 'ferrari',
             },
           },
         }}
+        id={tableId}
       />
     );
+    expect(screen.getByTestId(`${tableId}-table-toolbar-search`)).toBeInTheDocument();
+    expect(screen.getByTestId(`${tableId}-table-toolbar-search`)).toHaveValue('ferrari');
+  });
 
-    expect(wrapper.find(`.${prefix}--search-input`)).toHaveLength(1);
```

Comment:
Can we use the `rerender` method from testing library to see what happens when we update the defaultValue for search. I think this is something that we were asked to be able to support. I just want to be sure it's still functional.

Reply:
@davidicus Thank you for bringing this up. It was a bug. If the `defaultValue` updated then `TableToolbar` wouldn't rerender as well as table data.
It is fixed in latest commits by modifying `tableRegister` action, which is rerendering whole table in case of new defaultValue. Relevant test is added for this feature.
Also, it can be tested:

- in [this story](https://deploy-preview-3557--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-table--with-searching)
- Modify the `view.toolbar.search.defaultValue` input in the knobs area
- Verify that search input in toolbar is being rerendered and table is filtered accordingly

Reply:
Great!